# Project 3

In [57]:
import plotly.io as pio

pio.renderers.default = "vscode+jupyterlab+notebook_connected"

In [58]:
import pandas as pd
import plotly.express as px

In [59]:
df = pd.read_csv('/Users/yutonomy/Dropbox/Columbia SIPA/24FALL/Computing in Context/project3/Neighborhood.csv')
df

Year Published  PUMA Borough                               Neighborhoods  \
0              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
1              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
2              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
3              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
4              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
..              ...   ...     ...                                         ...   
380            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
381            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
382            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
383            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
384            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   

                           CD  Join  NYC_Poverty_Rate  Median_Income  \
0     BX Community District 8  3701             0.152          30437   
1     BX Community District 8  3701             0.152          30437   
2     BX Community District 8  3701             0.152          30437   
3     BX Community District 8  3701             0.152          30437   
4     BX Community District 8  3701             0.152          30437   
..                        ...   ...               ...            ...   
380  QN Community District 14  4114             0.179          25920   
381  QN Community District 14  4114             0.179          25920   
382  QN Community District 14  4114             0.179          25920   
383  QN Community District 14  4114             0.179          25920   
384  QN Community District 14  4114             0.179          25920   

     Perc_White  Perc_Black  ...  Ind6Rank  \
0         0.324       0.124  ...      32.0   
1         0.324       0.124  ...      38.0   
2         0.324       0.124  ...      27.0   
3         0.324       0.124  ...      41.0   
4         0.324       0.124  ...      28.0   
..          ...         ...  ...       ...   
380       0.350       0.353  ...      42.0   
381       0.350       0.353  ...      55.0   
382       0.350       0.353  ...      15.0   
383       0.350       0.353  ...       NaN   
384       0.350       0.353  ...       NaN   

                                        Ind6Definition  \
0    Percentage of jobs in small businesses (less t...   
1    Percentage of bank & credit union branches off...   
2    Number of pre-foreclosure notices issued per 1...   
3        Average (mean) travel time to work in minutes   
4    Percentage of people eligible for SNAP receivi...   
..                                                 ...   
380  Number of pre-foreclosure notices issued per 1...   
381      Average (mean) travel time to work in minutes   
382  Percentage of people eligible for SNAP receivi...   
383                                                NaN   
384                                                NaN   

                           Ind7 Ind7Outcome Ind7Rank  \
0       Participatory Budgeting    0.071777     26.0   
1    Mobile Banking Utilization    0.205800     23.0   
2          Financial Counseling    0.000000     40.0   
3                           NaN         NaN      NaN   
4        Tax Credit Utilization    0.966000      9.0   
..                          ...         ...      ...   
380        Financial Counseling    1.413969     21.0   
381                         NaN         NaN      NaN   
382      Tax Credit Utilization    0.936000     25.0   
383                         NaN         NaN      NaN   
384                         NaN         NaN      NaN   

                                        Ind7Definition  \
0    Percentage of eligible residents casting a vot...   
1    Estimated percentage of residents that used mo...   
2    Full-Time Equivalent Financial Counselors per ...   
3           

- **Dataset(s) to be used:** [https://data.cityofnewyork.us/Business/Neighborhood-Financial-Health-Digital-Mapping-and-/r3dx-pew9/about_data]
- **Analysis question:** The relationship between race and average income: is there any relationship between the percentage of black people and the change in the average income?
- **Columns that will (likely) be used:**
  - Median_Income
  - Perc_Black
- **Hypothesis**: The bigger the percentage of the black people gets, the less the average income becomes
- **Site URL:** [https://yutos-project3.readthedocs.io/en/latest/index.html]

# Check the data and clean the data if needed

In [60]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 385 entries, 0 to 384
Data columns (total 52 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Year Published    385 non-null    int64  
 1   PUMA              385 non-null    int64  
 2   Borough           385 non-null    object 
 3   Neighborhoods     385 non-null    object 
 4   CD                385 non-null    object 
 5   Join              385 non-null    int64  
 6   NYC_Poverty_Rate  385 non-null    float64
 7   Median_Income     385 non-null    int64  
 8   Perc_White        385 non-null    float64
 9   Perc_Black        385 non-null    float64
 10  Perc_Asian        385 non-null    float64
 11  Perc_Other        385 non-null    float64
 12  Perc_Hispanic     385 non-null    float64
 13  Goal              385 non-null    object 
 14  GoalName          385 non-null    object 
 15  GoalFullName      385 non-null    object 
 16  TotalOutcome      367 non-null    float64
 1

In [61]:
df.head()

Year Published  PUMA Borough                       Neighborhoods  \
0            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
1            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
2            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
3            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
4            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   

                        CD  Join  NYC_Poverty_Rate  Median_Income  Perc_White  \
0  BX Community District 8  3701             0.152          30437       0.324   
1  BX Community District 8  3701             0.152          30437       0.324   
2  BX Community District 8  3701             0.152          30437       0.324   
3  BX Community District 8  3701             0.152          30437       0.324   
4  BX Community District 8  3701             0.152          30437       0.324   

   Perc_Black  ...  Ind6Rank  \
0       0.124  ...      32.0   
1       0.124  ...      38.0   
2       0.124  ...      27.0   
3       0.124  ...      41.0   
4       0.124  ...      28.0   

                                      Ind6Definition  \
0  Percentage of jobs in small businesses (less t...   
1  Percentage of bank & credit union branches off...   
2  Number of pre-foreclosure notices issued per 1...   
3      Average (mean) travel time to work in minutes   
4  Percentage of people eligible for SNAP receivi...   

                         Ind7 Ind7Outcome Ind7Rank  \
0     Participatory Budgeting    0.071777     26.0   
1  Mobile Banking Utilization    0.205800     23.0   
2        Financial Counseling    0.000000     40.0   
3                         NaN         NaN      NaN   
4      Tax Credit Utilization    0.966000      9.0   

                                      Ind7Definition  \
0  Percentage of eligible residents casting a vot...   
1  Estimated percentage of residents that used mo...   
2  Full-Time Equivalent Financial Counselors per ...   
3                                                NaN   
4     Percentage of eligible Tax Units claiming EITC   

                              Ind8 Ind8Outcome  Ind8Rank  \
0                              NaN         NaN       NaN   
1                              NaN         NaN       NaN   
2                              NaN         NaN       NaN   
3                              NaN         NaN       NaN   
4  Adults With High School Diploma    0.845032      21.0   

                                      Ind8Definition  
0                                                NaN  
1                                                NaN  
2                                                NaN  
3                                                NaN  
4  Percentage of adults with a high school degree...  

[5 rows x 52 columns]

In [62]:
df.sample(10)

Year Published  PUMA    Borough  \
61             2020  3808  Manhattan   
63             2020  3709      Bronx   
343            2020  4108     Queens   
294            2020  4101     Queens   
342            2020  4108     Queens   
275            2020  4016   Brooklyn   
197            2020  4004   Brooklyn   
228            2020  4009   Brooklyn   
183            2020  4003   Brooklyn   
297            2020  4101     Queens   

                                   Neighborhoods                        CD  \
61       Murray Hill, Gramercy & Stuyvesant Town   MN Community District 6   
63       Castle Hill, Clason Point & Parkchester   BX Community District 9   
343                     Forest Hills & Rego Park   QN Community District 6   
294                   Astoria & Long Island City   QN Community District 1   
342                     Forest Hills & Rego Park   QN Community District 6   
275  Sheepshead Bay, Gerritsen Beach & Homecrest  BK Community District 15   
197               Brooklyn Heights & Fort Greene   BK Community District 2   
228                         Canarsie & Flatlands  BK Community District 18   
183                           Bedford Stuyvesant   BK Community District 3   
297                   Astoria & Long Island City   QN Community District 1   

     Join  NYC_Poverty_Rate  Median_Income  Perc_White  Perc_Black  ...  \
61   3808             0.100          73067       0.694       0.032  ...   
63   3709             0.264          20300       0.028       0.288  ...   
343  4108             0.163          41029       0.512       0.029  ...   
294  4101             0.183          31658       0.478       0.063  ...   
342  4108             0.163          41029       0.512       0.029  ...   
275  4016             0.194          29307       0.691       0.038  ...   
197  4004             0.121          44313       0.447       0.261  ...   
228  4009             0.150          32210       0.230       0.630  ...   
183  4003             0.226          24627       0.231       0.530  ...   
297  4101             0.183          31658       0.478       0.063  ...   

     Ind6Rank                                     Ind6Definition  \
61        NaN                                                NaN   
63       45.0      Average (mean) travel time to work in minutes   
343       NaN                                                NaN   
294      15.0  Percentage of bank & credit union branches off...   
342      41.0  Percentage of people eligible for SNAP receivi...   
275      40.0      Average (mean) travel time to work in minutes   
197       NaN                                                NaN   
228      52.0      Average (mean) travel time to work in minutes   
183      15.0  Percentage of jobs in small businesses (less t...   
297      43.0  Percentage of people eligible for SNAP receivi...   

                           Ind7 Ind7Outcome Ind7Rank  \
61                          NaN         NaN      NaN   
63                          NaN         NaN      NaN   
343                         NaN         NaN      NaN   
294  Mobile Banking Utilization    0.271100      9.0   
342      Tax Credit Utilization    0.917000     36.0   
275                         NaN         NaN      NaN   
197                         NaN         NaN      NaN   
228                         NaN         NaN      NaN   
183     Participatory Budgeting    0.082229     18.0   
297      Tax Credit Utilization    0.900000     40.0   

                                        Ind7Definition  \
61                                                 NaN   
63                                                 NaN   
343                                                NaN   
294  Estimated percentage of residents that used mo...   
342     Percentage of eligible Tax Units claiming EITC   
275                                                NaN   
197                                                NaN   
228                                              

# Data Visualization
We will do the data visualization: visualize in the bar graph and the scatterplot then figure out the relationship or correlation between % of black people and Median Incomes.

In [63]:
fig = px.bar(df, x='Perc_Black', y='Median_Income', title='the relationship between the race and median income', color_discrete_sequence=['#000000'])  # Pure black color 
fig.show()


In [64]:
fig = px.scatter(df, x='Perc_Black', y='Median_Income', title='the relationship between the race and median income')
fig.show()

# Reflection
Based on the bar graph and the scatterplot above, it can be said that the higher the percentage of Black people, the lower the median income tends to be, and conversely, the lower the percentage, the higher the median income.

# Check the regression

In [65]:
import numpy as np
x = df['Perc_Black']
y = df['Median_Income']

coefficients = np.polyfit(x, y, 1)

print(f"Regression coefficient (slope): {coefficients[0]}")
print(f"Intercept: {coefficients[1]}")

Regression coefficient (slope): -20276.573844372164
Intercept: 35800.935109920705


# Takeaway
The regression coefficient (slope): -20276.57 indicates that for every 1% increase in Perc_Black (percentage of the Black population), the Median_Income (median income) decreases by approximately $20,276. Since the value is negative, it suggests a negative linear relationship between Perc_Black and Median_Income.